src/framework/services/scroll/ScrollPosition.js

```diff
@@ -0,0 +1,65 @@
+
+import ThrottledEventDispatcher from '../event/ThrottledEventDispatcher';
+
+export default class ScrollPosition {
+
+  constructor() {
+    this.listeners = [];
+    this.previous = this.current = 0;
```

Comment:
That would be great. When I saw the examples I wondered where `distanceFromButton` was coming from. I thought maybe setting it to undefined in the constructor would make it clear that this is exposed on purpose.